securetea/lib/antivirus/tools/utils.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+u"""Utils module for SecureTea AntiVirus.
+
+Project:
+    ╔═╗┌─┐┌─┐┬ ┬┬─┐┌─┐╔╦╗┌─┐┌─┐
+    ╚═╗├┤ │  │ │├┬┘├┤  ║ ├┤ ├─┤
+    ╚═╝└─┘└─┘└─┘┴└─└─┘ ╩ └─┘┴ ┴
+    Author: Abhishek Sharma <abhishek_official@hotmail.com> , Jul 4 2019
+    Version: 1.4
+    Module: SecureTea
+
+"""
+
+import json
+import platform
+import hashlib
+import threading
+import subprocess
+import os
+
+
+# Create thread lock
+lock = threading.Lock()
+
+
+def json_to_dict(path):
+    """
+    Read JSON file into python dictionary.
+
+    Args:
+        path (str): Path of the JSON file
+
+    Raises:
+        None
+
+    Returns:
+        data (dict): JSON data formatted in Python dictionary
+    """
+    try:
+        with open(path, "r") as json_data_file:
+            data = json.load(json_data_file)
+            return data
+    except Exception as e:
+        print(e)
+
+
+def categorize_os():
+    """
+    Categorize operating system by its parent distribution.
+
+    Args:
+        None
+
+    Raises:
+        None
+
+    Returns:
+        None
+    """
+    os_name = get_system_name()
+    if os_name in ["ubuntu", "kali", "backtrack", "debian"]:
+        return "debian"
+    # elif some other OS, add their  name
+    else:  # if OS not in list
+        return None
+
+
+def get_system_name():
+    """
+    Return the name of the operating system.
+
+    Args:
+        None
+
+    Raises:
+        None
+
+    Returns:
+        os_name (str): Name of the operating system
+    """
+    os_name = platform.dist()[0]
+    return os_name.lower()
+
+
+def extractBytes(file_path):
+    """
+    Extracts and returns bytes of the file.
+
+    Args:
+        data (str): String data to encode
+
+    Returns:
+        bytes: Encoded data
+
+    Raises:
+        None
+    """
+    with open(file_path, "rb") as rf:
+        return rf.read()
+
+
+def get_md5_hash(file_path):
+    """
+    Return MD5 hash value of the file.
+
+    Args:
+        file_path (str): Path of the file
+
+    Raises:
+        None
+
+    Returns:
+        hash_value (str): MD5 hash value of the file
+    """
+    extracted_bytes = extractBytes(file_path)
+    hash_value = hashlib.md5(extracted_bytes).hexdigest()
```

Comment:
@rejahrehim @adeyosemanputra There is no other option rather than using MD5 hashes for our Hash Signature Scanner, as VirusShare only has MD5 hashes collections.
https://virusshare.com/hashes.4n6